master/networking/bgp.md

```diff
@@ -1,424 +1,172 @@
+{{site.prodname}}’s default behavior is to create a **full-mesh** of internal BGP (iBGP) connections where each node peers with each other. This design works great for small and medium-size deployments (< 100 nodes) in public cloud, but you can hit performance bottlenecks at around 100 nodes.
 
-To turn off the full BGP node-to-node mesh or to modify the global AS number, complete
-the following steps.
+To build large clusters in public cloud, **BGP route reflectors** can be used to reduce the number of BGP peerings used on each node. In this model, some nodes act as route reflectors and are configured to establish a full mesh amongst themselves. Other nodes are then configured to peer with a subset of those route reflectors.
 
-1. Issue the following command to determine if you have a `default` BGP configuration
-   resource.
+You can also run {{site.prodname}} on public cloud without BGP or route reflectors using {{site.prodname}}’s **VXLAN cross subnet capabilities**. For more information, see [Configure overlay networking]({{site.baseurl}}/{{page.version}}/networking/vxlan-ipip).
 
-    ```
-    calicoctl get bgpconfig default
-    ```
+#### Topologies for on-premises deployments
 
-1. If the resource _does_ exist, skip to step 3. Otherwise, use the following
-   command to create the resource. Before issuing the command, adjust the
-   `nodeToNodeMeshEnabled` and `asNumber` lines and values as desired.
-   Refer to [BGP Configuration Resource]({{site.baseurl}}/{{page.version}}/reference/resources/bgpconfig) for details about these settings.
+In **on-premises deployments**, you control the physical infrastructure, so you can configure {{site.prodname}} to peer directly with it. Typically, this involves disabling {{site.prodname}}’s default full-mesh behavior, and instead peer {{site.prodname}} with your L3 ToR router. There are many ways to build an on-premises BGP network. How you configure your autonomous systems is up to you - {{site.prodname}} works well with both iBGP and eBGP configurations. 
 
-    ```
-    cat << EOF | calicoctl create -f -
-    apiVersion: projectcalico.org/v3
-    kind: BGPConfiguration
-    metadata:
-      name: default
-    spec:
-      logSeverityScreen: Info
-      nodeToNodeMeshEnabled: false
-      asNumber: 63400
-    ```
+Depending on your topology, you may also consider using BGP route reflectors within each rack. However, this is typically needed only if the number of nodes in each L2 domain is large (> 100).
 
-    You're done!
+For a deeper look at common on-premises deployment models, see [Calico over IP Fabrics]({{site.baseurl}}/{{page.version}}/reference/architecture/design/l2-interconnect-fabric).
 
-1. If the resource _does_ exist, use the following command to retrieve it and save it
-   to a file.
+### Before you begin...
 
-    ```
-    calicoctl get bgpconfig default --export -o yaml > bgp.yaml
-    ```
+`calicoctl` must be installed and configured.
```

Comment:
Should we link to the calicoctl getting started page here?

Reply:
Yep, added link.